src/common/types/vector.cpp

```diff
@@ -139,6 +139,73 @@ void Vector::Initialize(const LogicalType &new_type, bool zero_data) {
 	}
 }
 
+struct DataArrays {
+	Vector &vec;
+	data_ptr_t data;
+	VectorBuffer *buffer;
+	idx_t type_size;
+	bool is_nested;
+	DataArrays(Vector &vec, data_ptr_t data, VectorBuffer *buffer, idx_t type_size, bool is_nested)
+	    : vec(vec), data(data), buffer(buffer), type_size(type_size), is_nested(is_nested) {};
+};
+
+void FindChildren(std::vector<DataArrays> &to_resize, VectorBuffer &auxiliary) {
```

Comment:
This looks quite similar to the other FindChildren below. Wonder whether this could be unified?

Reply:
The other one was dead code that I forgot to remove, thanks for noticing it :-)